CommonTools/RecoAlgos/interface/CosmicTrackingParticleSelector.h

```diff
@@ -70,7 +70,10 @@ class CosmicTrackingParticleSelector {
         minHit_(cfg.getParameter<int>("minHit")),
         chargedOnly_(cfg.getParameter<bool>("chargedOnly")),
         pdgId_(cfg.getParameter<std::vector<int> >("pdgId")),
-        beamSpotToken_(iC.consumes<reco::BeamSpot>(edm::InputTag("offlineBeamSpot"))) {}
+        beamSpotToken_(iC.consumes<reco::BeamSpot>(edm::InputTag("offlineBeamSpot"))),
+        trackerDigiGeomToken_(iC.esConsumes<TrackerGeometry, TrackerDigiGeometryRecord>()),
+        globalTrackingGeomToken_(iC.esConsumes<GlobalTrackingGeometry, GlobalTrackingGeometryRecord>()),
+        theMFToken_(iC.esConsumes<MagneticField, IdealMagneticFieldRecord>()) {}
```

Comment:
If I am not mistaken one does not need to re-declare the objects consumed in the tokens in the `esConsumes` call.  Removing them will make the code shorter.

Reply:
This is correct, plain `iC.esConsumes()` is sufficient (as would be `iC.consumes()` nowadays). They use a trick to deduce the product and record types from the token.